packages/redux/src/products/selectors/lists.ts

```diff
@@ -662,6 +663,8 @@ export const getHierarchicalFacetsWithChildren = createSelector(
       .map(id => {
         const facet = facets?.[id];
 
+        if (!facet) return null;
```

Comment:
From the type that was being inferred, we had: `(FacetEntity | ... | undefined)[]`.   
Considering my code change, now the type that is being inferred is `(FacetEntity | ... | null)[]`.   
It means that previously, `undefined` was an option and that's why the `.filter(Boolean)` was being applied. Currently, `undefined` is no longer an option; instead, `null` is now an option that will be filtered out by `filter(Boolean)`.

As you were saying, the final result will be the same since `.filter(Boolean)` is being applied. However, TS cannot infer that falsy values will be filtered out. 

If I change my code to `return;` instead of `return null` the signature will be the exact same as we had before my code change. Do you want me to change to that instead?